spec/models/periodic_expense_runner_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+describe PeriodicExpenseRunner, type: :model do
+  let(:runner) { PeriodicExpenseRunner.new }
+
+  context '#run' do
+    context 'monthly expense' do
+      context 'last paid over a month ago' do
+        it 'creates an expense' do
+          last_pay = Date.new(2015, 2, 1)
+          create(:periodic_expense, start_date: last_pay.prev_month, period: 'monthly', last_paid_at: last_pay)
+
+          Timecop.freeze(last_pay.next_month)
+
+          expect { runner.run }.to change { Expense.count }.by(1)
+        end
+      end
+
+      context 'last paid less than a month ago' do
+        it 'does not create an expense' do
+          last_pay = Date.new(2015, 2, 1)
+          create(:periodic_expense, start_date: last_pay.prev_month, period: 'monthly', last_paid_at: last_pay)
+
+          Timecop.freeze(last_pay.next_week)
+
+          expect { runner.run }.to change { Expense.count }.by(0)
```

Comment:
tip: this can also be written as:
`expect { runner.run }.not_to change { Expense.count }`

not too important, just a suggestion for readability